store/cachekv/mergeiterator.go

```diff
@@ -33,24 +33,8 @@ func newCacheMergeIterator(parent, cache types.Iterator, ascending bool) *cacheM
 }
 
 // Domain implements Iterator.
```

Comment:
```suggestion
// Domain implements Iterator.
// Returns parent domain because cache and parent iterator domains are the same.
```

Maybe we can add an explanation?